### DISCUSSION TITLE
Highlight bar in Stacked Bar Chart

### DISCUSSION THREAD

**Discussion by mfialko** [recharts/recharts#5131] [Q&A] [answered]
I want to highlight a bar on hover, but not the whole column. How to do this in recharts?

![Screen Recording 2024-10-20 at 15 12 41](https://github.com/user-attachments/assets/adde7595-fec2-4980-8781-77899086740a)

**Comment by ckifer** [comment, marked as answer]
In recharts versions < 2.10 you can set `Tooltip shared={false} />` and things should render similar to this.

After that though, there is a bug which prevents this https://github.com/recharts/recharts/issues/5124

**Reply by LexQzim** [reply]
> In recharts versions < 2.10 you can set `Tooltip shared={false} />` and things should render similar to this.
> 
> After that though, there is a bug which prevents this #5124

Unfortunatly this is not working. 
I use the recharts version 2.15 and I'm struggeling with this. 
My problem is that the `shared={false}` version highlights only the first bar so it stucks there. And also the animation is off.
I tried to create a custom component but unfortunatly there is no props which can show me which bar I'm currently hovering, or is it?

I read the other issue an hope that version 3 will fix it but now sure how long I have to wait. 
Currently I have a small workaround with out highlighting. Isn't perfect but functional.

**Reply by ckifer** [reply]
Not sure what you mean by the first bar is stuck?
here is this working in 2.15 https://codesandbox.io/p/sandbox/stacked-bar-chart-forked-kk2lxp without animation off

and here is the same with 3.0 alpha https://codesandbox.io/p/sandbox/stacked-bar-chart-forked-mzn4fp

**Reply by ckifer** [reply]
Oh yeah it is broken in 2.15 with `activeBar` which is the linked issue above. Try the 3.x sandbox there

**Reply by LexQzim** [reply]
The 3.0 alpha works nice. Would like to use it but I need to wait for the official release I guess. 

> Not sure what you mean by the first bar is stuck?
> here is this working in 2.15 https://codesandbox.io/p/sandbox/stacked-bar-chart-forked-kk2lxp without animation off

What I mean is, that when I set `shared:{false}` then the `content` prop of my customTooltip f.e. returns only the label of the first bar. The same is for the selected fill. I only fill the background of the first bar. 

Ok, then I will wait for a stable version.

**Reply by LexQzim** [reply]
OK, I couldn't wait. 
It works but it has a wierd behavior where the tooltips starts from the top left corner and flys to my cursor. You can also see it in your sandbox example. Else it works. Will test it a little bit.

And not after the update I got a `Maximum update depth exceeded.` error after hovering to ofter over the bars. The loading time also increased. Not sure if this comes from my custome components but in version 2 it works.

**Reply by PavelVanecek** [reply]
I think @ckifer fixed the flying tooltip in https://github.com/recharts/recharts/pull/5490 so that should be going out in next alpha I believe.

@LexQzim Can you pls share how to reproduce the `maximum update depth exceeded` problem? I will look into it. I just fixed one of these here: https://github.com/recharts/recharts/pull/5492 so that might or might not be the same.

**Reply by ckifer** [reply]
flying behavior fixed in alpha 5

**Reply by LexQzim** [reply]
> I think @ckifer fixed the flying tooltip in #5490 so that should be going out in next alpha I believe.
> 
> @LexQzim Can you pls share how to reproduce the `maximum update depth exceeded` problem? I will look into it. I just fixed one of these here: #5492 so that might or might not be the same.

Hey, I would like to give you an example but I' currently out of time. But will check out the new alpha asap.
